### PR TITLE
#1878 add staged actions support

### LINKED ISSUE #1878: Adding order edit to the jvm SDK > Add staged actions support

### PR DESCRIPTION
**Pull request by alex-ct**
# Description

Closes #1878 

# Checklist

- [ ] Update release Notes
- [ ] Add to the current github milestone 
- [ ] Update commercetools api reference

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/OrderMessage.java [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 @JsonTypeInfo(
         use = JsonTypeInfo.Id.NAME,
-        property = "type",
-        visible = true
+        property = "type"
+        //defaultImpl = SimpleOrderMessage.class
```
Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/OrderMessage.java]: Is this commented for debug purposes.

File: commercetools-models/src/test/java/io/sphere/sdk/orderedit/commands/stagedactions/OrderEditStagedActionsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+        testOrderEditStagedUpdateAction(setShippingAddressAndCustomShippingMethod);
+    }
+
+    private void testOrderEditStagedUpdateAction(final OrderEditStagedUpdateAction orderEditStagedUpdateAction) {
+        OrderEditFixtures.withUpdateableOrderEdit(client(), orderEdit -> {
+
+            final OrderEditUpdateCommand orderEditUpdateCommand = OrderEditUpdateCommand.of(orderEdit, AddStagedAction.of(orderEditStagedUpdateAction));
+            final OrderEdit updatedOrderEdit = client().executeBlocking(orderEditUpdateCommand);
+            Assertions.assertThat(updatedOrderEdit).isNotNull();
+            Assertions.assertThat(updatedOrderEdit.getStagedActions().get(0).getAction()).isEqualTo(orderEditStagedUpdateAction.getAction());
```
Comment (acbeni) [inline commercetools-models/src/test/java/io/sphere/sdk/orderedit/commands/stagedactions/OrderEditStagedActionsIntegrationTest.java]: Add a check that the `OrderEdit` has only one action.

**Comment by acbeni** [comment]
@alex-ct do you prefer to let the doc update as a last step ?

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/OrderMessage.java (hunk 1 above)
Comment (alex-ct) [reply commercetools-models/src/main/java/io/sphere/sdk/orders/messages/OrderMessage.java]: It is my mistake. It should be uncommented. It is used to deserialise messages not known to the jvm sdk.

**Comment by alex-ct** [comment]
> @alex-ct do you prefer to let the doc update as a last step ?

yes :)

**Pull request merged**